packages/swing-store/src/snapStore.js

```diff
@@ -5,6 +5,17 @@ import { createGzip, createGunzip } from 'zlib';
 import { assert, details as d } from '@agoric/assert';
 import { promisify } from 'util';
 
+/**
+ * @template T
+ * @typedef {{
+ *   has: (hash: string) => Promise<boolean>,
+ *   load: (hash: string, loadRaw: (filePath: string) => Promise<T>) => Promise<T>,
```

Comment:
I think it's better to templatize the load function only, not the whole SnapStore, to avoid having to be explicit for types throughout.

```suggestion
 *   load: <T>(hash: string, loadRaw: (filePath: string) => Promise<T>) => Promise<T>,
```